src/error.ts

```diff
@@ -5,14 +5,36 @@ import { AxiosError } from 'axios';
 export class CogniteError extends Error {
   public status: number;
   public requestId?: string;
-  constructor(errorMessage: string, status: number, requestId?: string) {
+  public missing?: object[];
+  public duplicated?: any[];
+  constructor(
+    errorMessage: string,
+    status: number,
+    requestId?: string,
+    extra: any = {}
+  ) {
     let message = `${errorMessage} | code: ${status}`;
     if (requestId) {
       message += ` | X-Request-ID: ${requestId}`;
     }
+    const { missing, duplicated } = extra;
+    if (missing && missing.length > 0) {
+      message += '\nMissing:';
+      missing.forEach((item: object) => {
+        message += `\n  ${JSON.stringify(item)}`;
+      });
+    }
+    if (duplicated && duplicated.length > 0) {
+      message += '\nDuplicated:';
+      duplicated.forEach((item: object) => {
+        message += `\n  ${JSON.stringify(item)}`;
```

Comment:
I suggest to pass the entire `extra` object in `stringify()` instead of "stringifying" it manually